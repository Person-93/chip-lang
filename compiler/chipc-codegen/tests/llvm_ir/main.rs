use chip_snapshot_tests::{assert_snapshot, cases, run_case};
use chipc_ast::Ast;
use chipc_codegen::CodeGen;
use cxx::let_cxx_string;

#[chip_test_harness::nextest_main(snapshot)]
fn main(name: &str) {
  let source = run_case(name);
  let ast = Ast::parse(&source);
  let hir = chipc_ast_lowering::lower("test-package".to_owned(), &ast);
  let mir = chipc_hir_lowering::lower(&hir);
  let mut code_gen = CodeGen::new("test-package", mir.root);
  let code_gen = code_gen.pin_mut();
  let code_gen = code_gen.as_ref();

  let_cxx_string!(output = "");
  code_gen.print_llvm_ir_to_string(output.as_mut());
  let output = output.to_str().unwrap().trim();

  let mut processed = String::new();
  for line in output.lines() {
    if line.strip_prefix("target datalayout = ").is_some() {
      processed += "target datalayout = \"<redacted>\"\n";
    } else if line.strip_prefix("target triple = ").is_some() {
      processed += "target triple = \"<redacted>\"\n";
    } else if line.strip_prefix("%Size = type { ").is_some() {
      processed += "%Size = type { [redacted] }\n";
    } else {
      processed += line;
      processed.push('\n');
    }
  }
  let processed = processed.trim();

  assert_snapshot!(processed);
}
