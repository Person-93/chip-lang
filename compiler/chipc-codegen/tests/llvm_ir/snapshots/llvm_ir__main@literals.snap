---
source: compiler/chipc-codegen/tests/llvm_ir/main.rs
assertion_line: 35
expression: processed
---
; ModuleID = 'test-package'
source_filename = "test-package"
target datalayout = "<redacted>"
target triple = "<redacted>"

%Size = type { [redacted] }

define internal void @main() {
entry:
  %var = alloca i32, align 4
  %var1 = alloca i8, align 1
  %var2 = alloca i16, align 2
  %var3 = alloca i32, align 4
  %var4 = alloca i64, align 8
  %var5 = alloca i128, align 8
  %var6 = alloca %Size, align 8
  %var7 = alloca i8, align 1
  %var8 = alloca i16, align 2
  %var9 = alloca i32, align 4
  %var10 = alloca i64, align 8
  %var11 = alloca i128, align 8
  %var12 = alloca %Size, align 8
  %var13 = alloca float, align 4
  %var14 = alloca double, align 8
  %var15 = alloca float, align 4
  %var16 = alloca float, align 4
  %var17 = alloca double, align 8
  %var18 = alloca i32, align 4
  %var19 = alloca i32, align 4
  %var20 = alloca { i8*, %Size }, align 8
  %var21 = alloca i1, align 1
  %var22 = alloca i1, align 1
  br label %bb

bb:                                               ; preds = %entry
  ret void
}
